### ISSUE TITLE
we need transport='websocket' with mono signalr 2.*  on linux

### ISSUE THREAD

**Issue by 564064202** [SignalR/SignalR#3969]
### Expected  behavior
we need transport='websocket' with mono signalr 2.*  on linux
### Actual behavior
doesn't support

**Comment by moozzyk** [comment]
Try using netstandard client shipped with SignalR 2.2. If Mono does not support ClientWebSocket natively you won't be able to use this transport in SignalR.

**Comment by 564064202** [comment]
we need transport='websocket' with mono signalr 2.* on linux
why ClientWebSocket ? server with 'websocket' transport does not work..

**Comment by 564064202** [comment]
@moozzyk

**Comment by moozzyk** [comment]
SignalR client does not contain implementation of websockets. It uses the implementation from the framework (which is ClientWebSockest). If the API/implementation is not available there is not much SignalR can do.

**Comment by 564064202** [comment]
@moozzyk  not client.
server

**Comment by moozzyk** [comment]
Same thing - the server uses WebSocket class which does not seem to be available in Mono: http://go-mono.com/status/status.aspx?reference=4.5&profile=4.5&assembly=System

**Comment by jacksonjim** [comment]
How to build the SignaIR client whith ARM Linux ? @moozzyk  the mono cannot support ARM CPU, so how to building SignalR client ?

**Comment by moozzyk** [comment]
SignalR client is pure managed code. The portable version supports the most platforms (has the least requirements/dependencies). If you don't have a runtime (CLR/Mono) you won't be able to run it however. Having said that - a few years ago [I tried running](https://blog.3d-logic.com/2013/05/19/raspberry-pi-mono-ef6-defeat/) Entity Framework on my Raspberry Pi. The one I used (Model A) is ARMv6 based and it I was able to install Mono on it, so I don't think it is true that Mono cannot support ARM.
